packages/frontend/src/hooks/usePositions.ts

```diff
@@ -72,23 +72,32 @@ export const usePositions = () => {
           const wethAmt = new BigNumber(isWethToken0 ? s.amount0 : s.amount1)
           const usdAmt = getUsdAmt(wethAmt, s.timestamp)
 
+          const time = new Date(Number(s.timestamp) * 1000).setUTCHours(0, 0, 0) / 1000
+
+          acc.ethCollateralPnl = acc.ethCollateralPnl.plus(
+            wethAmt.abs().times(new BigNumber(ethPriceMap[time]).minus(ethPrice)),
```

Comment:
Do we accumulate this under all swaps? if we partially close the short position, it seems to accumulate the wethamount even I am trying to close position
I think we only need to accumulate eth collateral amount for selling txs?
Or maybe we use abs() instead of using negated()?

Reply:
I actually think this works, it doesn't accumulate wethAmount when we partially close, it only calculates based on what's left in the position. Also, this calculation is only relevant when the user has a short position.

Reply:
But it seems to accumulate even when I have a bought swap, doesn't that mean to accumulate the eth collateral again when we partially close the position?

Reply:
Or even fully close the short position, for example, there will be two swaps, one is the selling tx, and the other is buying tx. We are going through both of them in this swaps iteration, but the eth collateral will accumulate only when we actually sell oSQTH. If we use abs() here, it means we make ethCollateral 2x bigger than it should be, cuz we accumulate it again

Reply:
I'm not sure I follow. Are you saying we are not supposed to accumulate the ethcollateral for partially closed positions? If yes, why? @yichiehliu

Reply:
Like this two txs
`{
        "__typename": "Swap",
        "amount0": "16.240200606076410471",
        "amount1": "-5.339246529671475668",
        "id": "0xc1b340861c4c831fa5c1b271a2c54bfda33aaa0291359da87cd66f097e32ec7f#667",
        "origin": "0x5261ae032506ab52f6b2592b3135cb1b7741d205",
        "recipient": "0x8903918dfe74476e90b63061e5b9c3e63b65d3f4",
        "timestamp": "1642598425",
        "transaction": {
            "__typename": "Transaction",
            "blockNumber": "11842156",
            "id": "0xc1b340861c4c831fa5c1b271a2c54bfda33aaa0291359da87cd66f097e32ec7f"
        }
    },
    {
        "__typename": "Swap",
        "amount0": "-16.240200606076410471",
        "amount1": "5.361107043676061641",
        "id": "0xe6078c740536792b862ea5c7a9e78f91bbaa0226cd9fadcbc11a10ef15341f9a#675",
        "origin": "0x5261ae032506ab52f6b2592b3135cb1b7741d205",
        "recipient": "0x8903918dfe74476e90b63061e5b9c3e63b65d3f4",
        "timestamp": "1642613814",
        "transaction": {
            "__typename": "Transaction",
            "blockNumber": "11843124",
            "id": "0xe6078c740536792b862ea5c7a9e78f91bbaa0226cd9fadcbc11a10ef15341f9a"
        }
    },`

These two is where I open a short position and fully close it
We are going through both of them in this swaps iteration, but we will get wethAmt.abs()  = 5.361107043676061641 in the first tx and get 5.339246529671475668 in the second tx, plus it again, cuz we use abs() here

Reply:
and acc.ethCollateralPnl will be 5.361107043676061641 x price changes + 5.339246529671475668 x price changes, but the ethCollateral is only 5.361107043676061641

Reply:
okay, I get it now @yichiehliu

Reply:
Also If they are just buying oSQTH, and then sell it
There wont be ethCollateralPnl at all since there is no ethCollateral in long scenario

Reply:
Also If they are just buying oSQTH, and then sell it, there wont be ethCollateralPnl at all since there is no ethCollateral under long scenario